lib/jsdom/level2/events.js

```diff
@@ -267,12 +265,33 @@ core.AttributeList.prototype.$setNode = attrNodeInterceptor(core.AttributeList.p
 core.AttributeList.prototype.$set = attrInterceptor(core.AttributeList.prototype.$set, true);
 core.AttributeList.prototype.$setNoNS = attrInterceptor(core.AttributeList.prototype.$setNoNS, false);
 
-core.Document.prototype.createEvent = function(eventType) {
-    switch (eventType) {
-        case "MutationEvents": return new core.MutationEvent(eventType);
-        case "UIEvents": return new core.UIEvent(eventType);
-        case "MouseEvents": return new core.MouseEvent(eventType);
-        case "HTMLEvents": return new core.Event(eventType);
+core.Document.prototype.createEvent = function(type) {
+    var Constructor = null;
+    var typeLower = type.toLowerCase();
+    var interfaceTable = {
```

Comment:
We can create this table outside the function so that callers don't have to create it every time.